amf-aml/shared/src/test/scala/amf/testing/parsing/DialectInstanceAssertionTest.scala

```diff
@@ -0,0 +1,47 @@
+package amf.testing.parsing
+
+import amf.aml.client.scala.model.document.DialectInstance
+import amf.core.client.common.position.Range
+import amf.aml.client.scala.{AMLConfiguration, AMLDialectInstanceResult}
+import amf.core.client.scala.model.domain.AmfArray
+import org.scalatest.funsuite.AsyncFunSuite
+import org.scalatest.matchers.should.Matchers
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class DialectInstanceAssertionTest extends AsyncFunSuite with Matchers {
+  override implicit val executionContext: ExecutionContext = ExecutionContext.Implicits.global
+  val basePath                                             = "file://amf-aml/shared/src/test/resources/vocabularies2/instances/"
+
+  def parseDialectInstance(dialectPath: String, instancePath: String): Future[AMLDialectInstanceResult] = {
+    AMLConfiguration.predefined().withDialect(dialectPath) flatMap { config =>
+      config.baseUnitClient().parseDialectInstance(instancePath)
+    }
+  }
+
+  def getDialectInstanceFirstValue(di: DialectInstance) =
+    di.encodes.fields
+      .fields()
+      .head
+      .value
+      .value
+      .asInstanceOf[AmfArray]
+      .values
+      .head
+
+  test("3604") {
```

Comment:
That's a short name for a test

Reply:
Split this test into two tests that assert that value. Asserting l1 == x, l2 == x is the same as asserting l1 == l2. It's weird asserting lexicals coming from from two different dialects are the same. Spliting will make each test "more unitary"

Reply:
agree, done